config/configload/load.go

```diff
@@ -76,89 +95,58 @@ func parseFile(filePath string, srcBytes *[][]byte) (*hcl.File, error) {
 		return nil, diags
 	}
 
-	return parsed, nil
+	body := parsed.Body.(*hclsyntax.Body)
+	err = absolutizePaths(body)
```

Comment:
Unfortunately it does not work in this early stage. There is no env readout and no `defaults.env...` handling across multiple files. Simplest case is `couper run -d ./public`. I have added an additional testcase to check this file too.

Reply:
Hmm, so you run `DOC_DIR=./www   couper run -f a/couper.hcl -f b/couper.hcl` and expect `DOC_DIR` to resolve to `a/www` **and** `b/www` at the same time depending on the context file?

Or you just want that error to disappear?

Reply:
First of all it should not brake our existing docker configuration. `env.` is not resolved at this stage while calling `absolutizePaths`. So the error is `public/couper.hcl:3,21-24: Variables not allowed; Variables may not be used here.`
But it should, especially with `defaults.environment_variables`. I have removed the NewContext call in c8a7e2ab3cf73b624a0712fa8ce9bcb04031cdf6 since it is shadowing the fact, that this call does not resolve `defaults.environment_variables` due to `nil` param.